src/utilities/include/par.h

```diff
@@ -49,47 +53,251 @@ inline constexpr ExecutionPolicy autoPolicy(size_t size) {
+          typename Dummy = void>
+void inclusive_scan(ExecutionPolicy policy, InputIter first, InputIter last,
+                    OutputIter d_first) {
+#if MANIFOLD_PAR == 'T'
+  if (policy == ExecutionPolicy::Par) {
+    tbb::parallel_scan(
+        tbb::blocked_range<size_t>(0, std::distance(first, last)),
+        static_cast<T>(0),
+        [&](const tbb::blocked_range<size_t> &range, T sum,
+            bool is_final_scan) {
+          T temp = sum;
+          for (size_t i = range.begin(); i < range.end(); ++i) {
+            temp = temp + first[i];
+            if (is_final_scan) d_first[i] = temp;
+          }
+          return temp;
+        },
+        std::plus<T>());
+    return;
   }
-#define STL_DYNAMIC_BACKEND_VOID(NAME)              \
-  template <typename... Args>                       \
-  void NAME(ExecutionPolicy policy, Args... args) { \
-    std::NAME(args...);                             \
+#endif
+  std::inclusive_scan(first, last, d_first);
+}
+
+#if MANIFOLD_PAR == 'T'
+template <typename T, typename InputIter, typename OutputIter, typename BinOp>
+struct ScanBody {
+  T sum;
+  T identity;
+  BinOp &f;
+  InputIter input;
+  OutputIter output;
+
+  ScanBody(T sum, T identity, BinOp &f, InputIter input, OutputIter output)
+      : sum(sum), identity(identity), f(f), input(input), output(output) {}
+  ScanBody(ScanBody &b, tbb::split)
+      : sum(b.identity),
+        identity(b.identity),
+        f(b.f),
+        input(b.input),
+        output(b.output) {}
+  template <typename Tag>
+  void operator()(const tbb::blocked_range<size_t> &r, Tag) {
+    T temp = sum;
+    for (size_t i = r.begin(); i < r.end(); ++i) {
+      T inputTmp = input[i];
+      if (Tag::is_final_scan()) output[i] = temp;
+      temp = f(temp, inputTmp);
+    }
+    sum = temp;
+  }
+  T get_sum() const { return sum; }
+  void reverse_join(ScanBody &a) { sum = f(a.sum, sum); }
+  void assign(ScanBody &b) { sum = b.sum; }
+};
+#endif
+
+template <typename InputIter, typename OutputIter,
+          typename BinOp = decltype(std::plus<typename std::iterator_traits<
+                                        InputIter>::value_type>()),
+          typename T = typename std::iterator_traits<InputIter>::value_type>
+void exclusive_scan(ExecutionPolicy policy, InputIter first, InputIter last,
+                    OutputIter d_first, T init = static_cast<T>(0),
+                    BinOp f = std::plus<T>(), T identity = static_cast<T>(0)) {
+#if MANIFOLD_PAR == 'T'
+  if (policy == ExecutionPolicy::Par) {
+    ScanBody<T, InputIter, OutputIter, BinOp> body(init, identity, f, first,
+                                                   d_first);
+    tbb::parallel_scan(
+        tbb::blocked_range<size_t>(0, std::distance(first, last)), body);
+    return;
+  }
+#endif
+  std::exclusive_scan(first, last, d_first, init, f);
+}
+
+// TODO: use STL variant when parallelization is not enabled?
```

Comment:
Looks like you already took care of this.

Reply:
Not completely. For example, for `transform_reduce`, I am calling `reduce` with transform iterator, which can be different from calling `std::transform_reduce`. Not sure about the performance impact of that, though.